telematic_system/scripts/log_analysis/README.md

```diff
@@ -0,0 +1,115 @@
+
+        ```
+    - Install python packages
+        ```
+        pip install -r requirements.txt
+        ```
+- Download the `log_timesheet.csv` to the current folder
+
+# Process V2xHub bridge log 
+1. Download v2xhub logs to the current folder.
+2. Run command to generate data publishing metrics.
+    ```
+    python3 parse_v2xhub_telematic_plugin_logs.py  --log_file_path <input-file-name>
+
+    e.g:
+    python3 parse_v2xhub_telematic_plugin_logs.py  --log_file_path T20_R6-13_V2XHub.log 
+    ```
+# Process Streets bridge log 
+1. Download streets bridge logs to the current folder.
+2. Run command to generate data publishing metrics.
+    ```
+    python3 parse_cloud_bridge_logs.py  <path-to-log-file>
+
+    e.g:
+    python3 parse_cloud_bridge_logs.py T20_R4_carma_cloud.log 
+    ```
+# Process Cloud bridge log 
+1. Download streets bridge logs to the current folder.
+2. Run command to generate data publishing metrics.
+    ```
+    parse_cloud_bridge_logs.py <path-to-log-file>
+
+    e.g:
+    python3 parse_cloud_bridge_logs.py  T20_R6-9_carma_cloud.log 
+    python3 parse_cloud_bridge_logs.py  T20_R10-13_carma_cloud.log 
+    ```
+
+# Process Vehicle bridge log 
+1. Download vehicle bridge logs to the current folder.
+2. Run command to generate data publishing metrics.
+    ```
+    python3 parse_vehicle_bridge_logs.py  <path-to-log-file>
+
+    e.g:
+    python3 parse_vehicle_bridge_logs.py T20_R6_R13_fusion/T20_R6_fusion.log 
+    ```
+# Process Messaging Server log 
+1. Download messaging server logs to the current folder.
+2. Run command to generate data publishing metrics.
+    ```
+    parse_messaging_server_logs.py <path-to-log-file>
+
+    e.g:
+    python3 parse_messaging_server_logs.py T20_R6-13_messaging_server.log
+    ```
+# Metric analysis
+## Latency
+1. Create a folder with the test case name.
+For example, test case 20:
+    ```
+    mkdir T20
+    ```
+2. Copy all the T20_*_messaging_server_*_delay_parsed.csv files to this new folder `T20`
+3. Run plot latency script to generate plots for those csv files with delay metrics in folder `T20`.
+    ```
+    python3 latencyPlotter.py <folder-name or test case name>
+
+    e.g:
+    python3 latencyPlotter.py T20
+    ``` 
+    The generated plots are saved into `output` folder.
+## Message loss
+1. Create a folder with the test case name and message drop.
+For example, test case 20:
+    ```
+    mkdir T20_message_drop
+
+    ```
+2. Copy all  <test case name>_*_messaging_server_*_message_drop_parsed.csv files to this new folder `<test case name>_message_drop`.
+3. Run message drop analysis script to analyze all files in the `<test case name>_message_drop` folder.
+    ```
+    python3 get_message_drop.py <folder-name or test case name>_message_drop
+
+    e.g:
+    python3 get_message_drop.py T20_message_drop
+    ```
+Generated result is similar to below:
+![Message_loss_result](https://github.com/user-attachments/assets/15fefacb-e929-4340-a0e3-6d7f6441ba8e)
```

Comment:
Need a new line before this image if you say it's "below".  See https://github.com/usdot-fhwa-stol/cda-telematics/tree/hotfix_analysis_script/telematic_system/scripts/log_analysis , the image is showing beside the ":".